base_layer/core/src/transactions/transaction_components/transaction_output.rs

```diff
@@ -261,6 +261,19 @@ impl TransactionOutput {
             &self.encrypted_data,
             self.minimum_value_promise,
         );
+        // Let's first verify that the metadata signature is valid
```

Comment:
Might be worth noting that this is (hopefully) an extra check that should also be done elsewhere, and is being done here for added safety.